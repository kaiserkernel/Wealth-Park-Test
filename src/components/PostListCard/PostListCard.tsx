import { FC } from 'react'

import { FaCalendar } from 'react-icons/fa'
import { format } from 'date-fns'
import { IPost } from '@src/types'

interface Props {
  post: IPost
}

const PostListCard: FC<Props> = ({ post }) => {
  return (
    <a
      target="_blank"
      href={post.url}
      className="flex flex-col rounded overflow-hidden border mb-4"
      data-testid="post-list-card"
      rel="noreferrer"
    >
      <div className="flex flex-row items-center p-2">
        <FaCalendar />
        <p className="ml-2">{format(new Date(post.time * 1000), 'MM/dd/yyyy')}</p>
        <p className="font-bold text-center p-2">{post.title}</p>
      </div>
      {/* <p className="p-2">{post.text}</p> */}
    </a>
  )
}

export default PostListCard
