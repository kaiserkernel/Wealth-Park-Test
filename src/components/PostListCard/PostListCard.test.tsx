import { render, screen } from '@testing-library/react'

import PostListCard from './PostListCard'

describe('Post List Card', () => {
  const POST = {
    post: {
      by: 'mmastrac',
      descendants: 10,
      id: 34783447,
      kids: [34784221, 34784342, 34784198, 34783951, 34783941],
      score: 77,
      time: 1676335246,
      title: 'Google’s Fully Homomorphic Encryption Compiler – A Primer',
      type: 'story',
      url: 'https://jeremykun.com/2023/02/13/googles-fully-homomorphic-encryption-compiler-a-primer/',
    },
  }

  it('renders correctly', () => {
    const { container } = render(<PostListCard {...POST} />)
    expect(container).toMatchSnapshot()
  })

  it('renders PostListCard', () => {
    render(<PostListCard {...POST} />)

    const postListCard = screen.getByTestId('post-list-card')

    expect(postListCard).toBeInTheDocument()
  })
})
