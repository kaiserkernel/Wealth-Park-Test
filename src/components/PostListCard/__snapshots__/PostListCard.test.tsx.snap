// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Post List Card renders correctly 1`] = `
<div>
  <a
    class="group rounded overflow-hidden border bg-gray-700"
    data-testid="post-list-card"
    href="/post/1"
  >
    <div
      class="h-[200px] relative"
    >
      <span
        style="box-sizing: border-box; display: block; overflow: hidden; background: none; opacity: 1; border: 0px; margin: 0px; padding: 0px; position: absolute; top: 0px; left: 0px; bottom: 0px; right: 0px;"
      >
        <img
          alt=""
          class="object-center object-cover transition duration-300 transform group-hover:scale-110 "
          data-nimg="fill"
          decoding="async"
          src="data:image/gif;base64,R0lGODlhAQABAIAAAAAAAP///yH5BAEAAAAALAAAAAABAAEAAAIBRAA7"
          style="position: absolute; top: 0px; left: 0px; bottom: 0px; right: 0px; box-sizing: border-box; padding: 0px; margin: auto; display: block; width: 0px; height: 0px; min-width: 100%; max-width: 100%; min-height: 100%; max-height: 100%; filter: blur(20px); background-size: cover; background-image: url(data:image/png;base64,iVBORw0KGgoAAAANSUhEUgAAAAEAAAABCAQAAAC1HAwCAAAAC0lEQVR42mNcWg8AAc8BJpg2zxQAAAAASUVORK5CYII=); background-position: 0% 0%;"
        />
        <noscript />
      </span>
    </div>
    <p
      class="font-bold text-center p-2"
    />
  </a>
</div>
`;
